### PR TITLE
Oauth2: Marking token holding cookie as httpOnly to prevent attacks

### PR DESCRIPTION
**Pull request by ktn-nv**
Commit Message:
OAuth2: Mark cookie, which holds tokens, as httpOnly

Additional Description:
Oauth2 filter sets the tokens in cookie but it doesn't mark it as httpOnly. This makes the applications open to attacks since the tokens can be stolen by injected JavaScript.

Risk Level: HIGH
Testing: There is no impact to the behavior of the OAuth2 filter. This only fixes the security of the cookie on the browser side.
Fixes: https://github.com/envoyproxy/envoy/issues/24097

### CODE REVIEW

**Comment by yanavlasov** [comment]
Since this is a user visible behavior change, it will need to be protected with the temporary runtime flag. Please see #26764 for an example.

**Comment by yanavlasov** [comment]
Thanks you for making this change. Please apply this diff to fix format https://dev.azure.com/cncf/envoy/_build/results?buildId=136778&view=logs&j=c7de00aa-b0fb-52ba-707b-da6afd918eda&t=294dc2ce-3369-5e4f-90de-3ad0f72952a1&l=94

**Comment by yanavlasov** [comment]
Please fix DCO. Here are the instructions https://github.com/envoyproxy/envoy/pull/27297/checks?check_run_id=13391344504

File: source/extensions/filters/http/oauth2/filter.cc [hunk 1]
```diff
@@ -535,18 +532,26 @@ void OAuth2Filter::addResponseCookies(Http::ResponseHeaderMap& headers,
   // If opted-in, we also create a new Bearer cookie for the authorization token provided by the
   // auth server.
   if (config_->forwardBearerToken()) {
+    std::string cookie_attribute_httponly = cookie_tail_http_only;
```
Comment (yanavlasov) [inline source/extensions/filters/http/oauth2/filter.cc:535]: nit: the following 6 lines can be replaced with ternary operator for readablity.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
The change looks good module a small nit. However DCO still needs to be fixed (otherwise the build fails). 

/wait

File: source/extensions/filters/http/oauth2/filter.cc (hunk 1 above)
Comment (ktn-nv) [reply source/extensions/filters/http/oauth2/filter.cc]: Done.

**Review by yanavlasov — APPROVED** [review]
LGTM from me. Adding owners for final pass.

**Comment by yanavlasov** [comment]
@derekargueta @snowp for final review

**Pull request merged**